src/ucp/api/ucp.h

```diff
@@ -660,16 +741,27 @@ typedef struct ucp_ep_params {
     uint64_t                field_mask;
 
     /**
-     * Destination address; the address must be obtained using
-     * @ref ucp_worker_get_address.
-     * This field is mandatory for filling (along with its corresponding bit
-     * in the field_mask - UCP_EP_PARAM_FIELD_REMOTE_ADDRESS).
-     * The ucp_ep_create routine will return with an error if the address isn't
+     * Destination address; if a client-server connection establishment
```

Comment:
ucp_address_unpack fails means the parameter is invalid, not that "it was not provided"
Before the change it was `(addr_not_specified) -> error`
After your change it is `(FIELD_REMOTE_ADDRESS_is_set && addr_not_specified) -> error`
The correct logic is `(FLAGS_CLIENT_SERVER_is_not_set && addr_not_specified) -> error`